src/Install.php

```diff
@@ -487,6 +490,57 @@ protected static function create_notes() {
 		WC_Admin_Notes_Historical_Data::possibly_add_note();
 	}
 
+	/**
+	 * Update selected notes, if they need changing from when they were
+	 * originally created.
+	 */
+	protected static function update_notes() {
```

Comment:
It is not clear to me at all what this function is doing. It appears that it is just copying existing notes, setting the same values as previously existed. I must be missing something! Please help me understand this better by updating naming and/or comments to make it very clear why the notes need changing, and what is being done.

Reply:
>  It appears that it is just copying existing notes, setting the same values as previously existed

That's pretty much it :-). At the moment if you change the note factory it doesn't flow through to the database. So recently I made a change to `WC_Admin_Notes_Deactivate_Plugin` so that the note that gets generated has a type of `info` rather than `update` (so that the note is dismissible), however that only works for new instances of the note.

This PR addresses that by creating a new instance of the note from the factory, assigning the database note ID to that note (so that it will overwrite the DB note on save), and copying some values from the database so it maintains state (date added, soft-delete status, etc).